src/renderer/runtime.ts

```diff
@@ -903,12 +921,21 @@ export class Runtime {
       // @todo add option in renderer to omit this transform, instead passing it as a param.
       const position = transformation ? transform(point, transformation, this.transformBuffer) : point;
       // Another hook before painting a layer.
+
+      const center = this.viewerToWorld(this.width / 2, this.height /2);
```

Comment:
@stephenwf I could use a little help/advice on this.  I was expecting this this method to work consistently, but it seems to be producing unexpected positions in CanvasPanel, do you have any ideas how to convert the center of the viewport to the center of the world?

Reply:
Okay, I think the issue is the `this.width` and `this.height` properties. They are not the height of the viewport. The viewport is sort of like an annotation thats rendered onto the image, and then the renderer will draw that somewhere at a specific size. So the `this.width` and `this.height` are "world" co-ordinates. 

The viewport height and width can be retrieved from the renderer:

```ts
const position = this.renderer.getRendererScreenPosition();

position.width // w
position.height // h
```

Reply:
Oh me too. I stared at that for a good while!